neuralmonkey/encoders/cnn_encoder.py

```diff
@@ -172,50 +155,8 @@ def __init__(self, data_id, convolutions, rnn_layers,
                             last_layer, keep_prob=self.dropout_placeholder)
 
                 last_layer = last_layer * last_padding_masks
-            last_layer_size = last_n_channels * image_height * image_width
-
-            with tf.variable_scope("rnn_inputs"):
-                encoder_ins = [
-                    tf.reshape(x, [-1, last_n_channels * image_height])
-                    for x in tf.split(2, image_width,
-                                      last_layer, name='split_input')]
-
-            def rnn_encoder(inputs, last_layer_size, scope):
-                with tf.variable_scope(scope):
-                    encoder_layers = []
-                    for size in rnn_layers:
-                        cell_g = tf.nn.rnn_cell.GRUCell(size, last_layer_size)
-                        last_layer_size = size
-                        cell = tf.nn.rnn_cell.DropoutWrapper(
-                            cell_g, output_keep_prob=self.dropout_placeholder)
-                        encoder_layers.append(cell)
-
-                    encoder_cell = tf.nn.rnn_cell.MultiRNNCell(encoder_layers)
-                    last_layer_size = len(encoder_layers) * last_layer_size
-                    # MultiRNNCell concatenates output of all the recurent
-                    # layers,but we want only the very last one
-                    _, encoder_state_concatenated = tf.nn.rnn(
-                        encoder_cell, inputs, dtype=tf.float32)
-
-                    if concatenate_rnns:
-                        encoder_state = encoder_state_concatenated
-                    else:
-                        encoder_state = encoder_state_concatenated[
-                            :, sum(rnn_layers[:-1]):]
-                    return encoder_state
-
-            encoder_state = rnn_encoder(
-                encoder_ins, last_layer_size, "encoder-forward")
-
-            # pylint: disable=redefined-variable-type
-            if bidirectional:
-                backward_encoder_state = rnn_encoder(
-                    list(reversed(encoder_ins)),
-                    last_layer_size, "encoder-backward")
-                encoder_state = tf.concat(
-                    1, [encoder_state, backward_encoder_state])
-
-            self.encoded = encoder_state
+
+            self.encoded = tf.reduce_mean(last_layer, [2, 3])
```

Comment:
Ten komentář pomoh hlavně mě, protože to průměruju přes blbý dimenze samozřejmě :-D